kale/prepdata/supergraph_construct.py

```diff
@@ -0,0 +1,250 @@
+
+        self.direction = (source_supervertex, target_supervertex)
+        self.source_supervertex = source_supervertex
+        self.target_supervertex = target_supervertex
+        self.edge_index = edge_index
+        self.edge_weight = edge_weight
+
+    def __repr__(self) -> str:
+        return f"SuperEdges(\n    edge_direction={self.source_supervertex}->{self.target_supervertex}, \n    edge_index={self.edge_index.shape})"
+
+
+class SuperVertexParaSetting(object):
+    r"""Parameter settings for each supervertex.
+
+        Args:
+            supervertex_name (str): the name of the supervertex.
+            inter_feat_dim (int): the dimension of the output of the internal feature layer.
+            inter_agg_dim (List[int]): the output dimensions of a sequence of internal aggregation layers.
+            exter_agg_dim (Dict[str, int], optional): the dimension of received message vector from parient supervertices.
+                Defaults to None.
+            mode (str, optinal): the allowed gripnet mode--'cat' or 'add'. Defaults to None.
+            num_bases (int, optional): Number of bases used for basis-decomposition if the supervertex is multi-relational. Defaults to 32.
+            if_catout (bool, optional): if concatenate the output of each layers. Defaults to True.
+        """
+
+    def __init__(
+        self,
+        supervertex_name: str,
+        inter_feat_dim: int,
+        inter_agg_dim: List[int],
+        exter_agg_dim: Dict[str, int] = None,
+        mode: str = None,
+        num_bases: int = 32,
+        if_catout: bool = True,
+    ) -> None:
+        self.supervertex_name = supervertex_name
+        self.inter_feat_dim = inter_feat_dim
+        self.inter_agg_dim = inter_agg_dim
+        self.mode = mode
+        self.num_bases = num_bases
+        self.if_catout = if_catout
+        self.exter_agg_dim = exter_agg_dim
+
+        # check if the mode is valid
+        if self.mode is not None and self.mode not in ["cat", "add"]:
+            error_msg = "The mode is not valid. It should be 'cat' or 'add'."
+            logging.error(error_msg)
+            raise ValueError(error_msg)
+
+
+class SuperGraph(object):
+    r"""
+    The supergraph structure in GripNet. Each supergraph is a directed acyclic graph (DAG) containing
+    supervertices and superedges.
+
+    Args:
+        supervertex_list (list[SuperVertex]): a list of supervertices.
+        superedge_list (list[SuperEdge]): a list of superedges.
+        supervertex_para_setting (dict[str, SuperVertexParaSetting], Optional): the parameter settings for each supervertex.
+    """
+
+    def __init__(
+        self,
+        supervertex_list: List[SuperVertex],
+        superedge_list: List[SuperEdge],
+        supervertex_setting_dict: Dict[str, SuperVertexParaSetting] = None,
+    ) -> None:
+
+        self.supervertex_dict = {sv.name: sv for sv in supervertex_list}
+        self.superedge_dict = {se.direction: se for se in superedge_list}
```

Comment:
`sv` `se`: better to make them more readable?

Reply:
If yes, change all.

Reply:
Changed in [888961a](https://github.com/pykale/pykale/pull/338/commits/888961ae2461a480da9cdc1524498781db773457).